applications/GeoMechanicsApplication/custom_workflows/time_loop_executor_interface.h

```diff
@@ -0,0 +1,38 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//                   Anne van de Graaf
+//                   Wijtze Pieter Kikstra
+//
+
+#pragma once
+
+#include <vector>
+#include <memory>
+
+#include "time_step_end_state.hpp"
+#include "time_incrementor.h"
+#include "strategy_wrapper.hpp"
+
+namespace Kratos
+{
+
+class Process;
+
+class TimeLoopExecutorInterface{
```

Comment:
@avdg81 We discussed we didn't want Interface in the name, but @WPK4FEM and I could not think of a suitable pair of names for the interface + implementation of the time loop executor. If you have a suggestion, please let us know!